text/0026-semantic-versioning.md

```diff
@@ -0,0 +1,189 @@
+|-|-|-|
+| \<Any incompatible change\> | \<Feature additions but compatible with previous release\> | \<Fixes but no major feature changes\> |
+| Change in Buildpack ID | Support for a new stack | Bug fixes |
+| Change in Buildpack API version | Buildpack runs a new command the causes backward compatible<br>change in behavior (e.g. x-install buildpack adds running "x clean") | Performance improvement, internal tweaks |
+| Change in API (Provides/Requires) | New configuration option (e.g. new environment variable support) | CVE fixes |
+| Removal of a stack or renaming of an existing stack | Change in <br>[buildpack-specific metadata](https://github.com/buildpacks/spec/blob/buildpack/v0.6/buildpack.md#build-plan-toml) | Change of required build tool version to package buildpack<br>(e.g. go version) |
+| Removal of configuration options<br>(e.g. buildpack.yml) |  |  |
+<br/>
+
+#### Buildpack Dependencies
+
+For buildpacks whose primary function is to provide dependencies, the following versioning is expectated
+when versions of dependencies are added or removed. The table below assumes that upstream dependencies
+follow [semantic versioning](https://semver.org/). For those that do not, maintainers may determine
+their versioning independently.
+
+| Dependency change | Buildpack Version expectation |
+|-|-|
+| ADD Patch | Bump Patch |
+| ADD Minor | - Bump Minor if set as default<br>  version<br>- Bump Patch otherwise |
+| ADD Major | - Bump Major if set as default<br>  version<br>- Bump Minor otherwise |
+| REMOVE Patch | Bump Patch |
+| REMOVE Minor | - Bump Major if unset as default<br>  version<br>- Bump Minor otherwise |
+| REMOVE Major | Bump Major |
+<br/>
+
+
+#### Composite Buildpack
+| `Major` | `Minor` | `Patch` |
+|-|-|-|
+| Implementation buildpack major bumps | Implementation buildpack minor bumps | Implementation buildpack patch bumps |
+| Removal of an order group dropping<br>support of some workflow | Addition of an order group that do<br>not cause change in detection logic |  |
+| Swapping order of order groups |  |  |
+| Addition of an order group<br>causing change in detection logic |  |  |
+<br/>
+
+
+#### Builder
+| `Major` | `Minor` | `Patch` |
+|-|-|-|
+| Lifecycle major bumps | Minor bumps of components | Patch bumps of components |
+| Stack Image major bumps | Addition of a buildpack<br>as last in the order |  |
+| Buildpack major bumps |  |  |
+| Change in Stack ID |  |  |
+| Removal of a Buildpack |  |  |
+| Swapping order of buildpacks |  |  |
+| Addition of a buildpack except<br>as last in the order |  |  |
+<br/>
+
+#### Stack
+| `Major`                 | `Minor`                   | `Patch`                   |
+|-------------------------|---------------------------|---------------------------|
+| Removal of  a package   | Addition of a new package | Update of a package       |
+
+### Intial Development
+
+A version of `0.0.x` is considered to be in initial development and may not be
+suitable for user consumption.
+
```

Comment:
Might be worth calling out that for alpha versions (0.0.x) all changes are patchs and for beta versions (0.x.y) breaking changes (listed as major versions above) will be minor bumps.

Reply:
My idea was that - 0.0.x do not follow any of these guidelines as they are not yet advertised as suitable for general consumption, and when you release 0.1.0 or higher you will abide by these semver guidelines, i.e. an immediate breaking change after 0.1.0 should be 1.0.0. Happy to change this per your comment if you feel strongly

Reply:
@ekcasey (friendly ping to see if this has to be changed)

Reply:
`1.0.0` indicates a level of maturity and readiness for production use cases. I wouldn't want us to move to `1.0.0` just because we wanted to make a breaking change. Yes we could use 0.0.x for this but 0.0.x basically communicates "experimental/unstable do not use". Whereas an "0.x.y" version is more stable and robust ready for feedback from early adopters.

Semver docs seem to align with making braking changes in an `0.y.0` version:

>Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.